### PR TITLE
added safe versions of minimum and maximum for List

### PR DESCRIPTION
**Pull request by shimib**
Provided safe minimumOption and maximumOption methods to List.

### CODE REVIEW

File: core/src/main/java/fj/data/List.java [hunk 1]
```diff
@@ -1466,6 +1466,16 @@ public final A maximum(final Ord<A> o) {
   }
 
   /**
+   * Returns the maximum element in this list according to the given ordering.
+   *
+   * @param o An ordering for the elements of the list.
+   * @return The optional maximum element in this list according to the given ordering.
+   */
+  public final Option<A> maximumOption(final Ord<A> o) {
+	  return headOption().map(x -> maximum(o));
```
Comment (jbgi) [inline core/src/main/java/fj/data/List.java]: should be space only indentation.

**Comment by shimib** [comment]
Sure
Will do

**Comment by shimib** [comment]
changed. Plz let me know if you have any comments.

**Comment by jbgi** [comment]
thanks!

**Pull request merged**